gateway/src/lib.rs

```diff
@@ -169,6 +176,60 @@ impl<'de> Deserialize<'de> for AccountName {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, sqlx::Type, Serialize)]
+#[sqlx(transparent)]
+pub struct Fqdn(String);
```

Comment:
Yea, that's a good idea. It looks like the only downside will be needing to manually implement `sqlx::Type` and `serde::Serialize` since `fqdn::FQDN` doesn't implement any of them itself.